framework/db/BaseActiveRecord.php

```diff
@@ -418,7 +418,13 @@ public function getRelatedRecords()
      */
     public function hasAttribute($name)
     {
-        return isset($this->_attributes[$name]) || in_array($name, $this->attributes());
+        static $attributes = [];
+
+        if (!array_key_exists($name, $attributes)) {
+            $attributes[$name] = isset($this->_attributes[$name]) || in_array($name, $this->attributes());
+        }
+
+        return $attributes[$name];
```

Comment:
No, it not different. `ActiveRecord::attributes()` are bound to the table schema by default, but this does not mean someone can not override and adjust this method.